src/Ratchet/WebSocket/WsServer.php

```diff
@@ -61,6 +61,13 @@ class WsServer implements HttpServerInterface {
      */
     private $msgCb;
 
+    /**
+     * Storage for dynamic properties.
+     *
+     * @var array
+     */
+    protected $_properties = [];
```

Comment:
Seeing this is reused in a lot of places, this should be a trait or `AllowDynamicProperties` can be used instead

Reply:
`#[AllowDynamicProperties]` is only available after 8.0 but traits may be a better option

Reply:
That's not an issue, php attributes are made to be backwards compatible all the way to php 4, because they are treated as comments otherwise and an attribute class that doesn't exist is simply ignored, this was thought specifically so that DynamicProperties support could be kept by only adding this attribute without being a breaking change

Reply:
I didn't think about that, I've already switched it to a trait.